src/modm/io/module.lb

```diff
@@ -37,6 +38,9 @@ def prepare(module, options):
             BooleanOption(name="with_printf",
                           description="Support for printf-style formatting",
                           default=not is_avr))
+    module.add_option(
```

Comment:
Can you help me understand this? What's the compilation bug? And why is `is_windows` set to `is_hosted`?

Reply:
FWIW, we should try to make the distinction between windows-specific fixes, hosted fixes and test environment. We really want this to work on Linux also.

Reply:
The compilation bug was multiple definitions of printf, so I didn't include `stdio.h`. I'm not sure if this is a windows-specific bug or not

Reply:
I'll call it a hosted-specific bug until I test building this module with hosted-linux

Reply:
It might be worth removing that include entirely and seeing if it still compiles for our normal stm32 built. It might be completely unnecessary.

Otherwise, it would probably be best to edit the comment anywhere there was some sort of compiler error to say what exactly it was (ideally including error message) so it's clear. Right now I can't tell from these comments what the issue was or what the change was.

Reply:
OK, commented out the include and included the error message I receive when I compile it with the include in the source code so we know why that's commented out.